src/results.cpp

```diff
@@ -370,38 +370,42 @@ util::Optional<Mixed> Results::aggregate(size_t column, bool return_none_for_emp
-                     [=](auto const& table) { return table.maximum_timestamp(column); });
+    size_t return_ndx;
+    auto results = aggregate(column, "max",
+            [&](auto const& table) { return table.maximum_int(column, &return_ndx); },
+            [&](auto const& table) { return table.maximum_float(column, &return_ndx); },
+            [&](auto const& table) { return table.maximum_double(column, &return_ndx); },
+            [&](auto const& table) { return table.maximum_timestamp(column, &return_ndx); });
+    return return_ndx == npos ? none : results;
 }
 
 util::Optional<Mixed> Results::min(size_t column)
 {
-    return aggregate(column, true, "min",
-                     [=](auto const& table) { return table.minimum_int(column); },
-                     [=](auto const& table) { return table.minimum_float(column); },
-                     [=](auto const& table) { return table.minimum_double(column); },
-                     [=](auto const& table) { return table.minimum_timestamp(column); });
+    size_t return_ndx;
+    auto results = aggregate(column, "min",
+            [&](auto const& table) { return table.minimum_int(column, &return_ndx); },
+            [&](auto const& table) { return table.minimum_float(column, &return_ndx); },
+            [&](auto const& table) { return table.minimum_double(column, &return_ndx); },
+            [&](auto const& table) { return table.minimum_timestamp(column, &return_ndx); });
+    return return_ndx == npos ? none : results;
 }
 
 util::Optional<Mixed> Results::sum(size_t column)
 {
-    return aggregate(column, false, "sum",
-                     [=](auto const& table) { return table.sum_int(column); },
-                     [=](auto const& table) { return table.sum_float(column); },
-                     [=](auto const& table) { return table.sum_double(column); },
-                     [=](auto const&) -> util::None { throw UnsupportedColumnTypeException{column, m_table, "sum"}; });
+    return aggregate(column, "sum",
+            [=](auto const& table) { return table.sum_int(column); },
+            [=](auto const& table) { return table.sum_float(column); },
+            [=](auto const& table) { return table.sum_double(column); },
+            [=](auto const&) -> util::None { throw UnsupportedColumnTypeException{column, m_table, "sum"}; });
 }
 
 util::Optional<Mixed> Results::average(size_t column)
 {
-    return aggregate(column, true, "average",
-                     [=](auto const& table) { return table.average_int(column); },
-                     [=](auto const& table) { return table.average_float(column); },
-                     [=](auto const& table) { return table.average_double(column); },
-                     [=](auto const&) -> util::None { throw UnsupportedColumnTypeException{column, m_table, "average"}; });
+    return aggregate(column, "average",
+            [=](auto const& table) { return table.is_empty() ? none : Optional<Mixed>(table.average_int(column)); },
```

Comment:
Why is it necessary to explicitly check for an empty table? Shouldn't `Table::average_foo` / `TableView::average_foo` be handling the empty case themselves?